test/simulator/items/shedshell.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const assert = require('assert');
+let battle;
+
+describe('Shed Shell', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should allow a Pokemon to escape a trapping abilities', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: "Gothitelle", ability: 'shadowtag', moves: ['calmmind']}]);
+		battle.join('p2', 'Guest 2', 1, [
+			{species: "Starmie", ability: 'naturalcure', item: 'shedshell', moves: ['recover']},
+			{species: "Heatran", ability: 'flashfire',  moves: ['rest']}
+		]);
+		battle.choose('p2', 'switch 2');
+		battle.commitDecisions();
+		assert.strictEqual(battle.p2.active[0].speciesid, 'heatran');
+	});
+
+	it('should allow a Pokemon to escape from moves that would trap them', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: "Gengar", ability: 'levitate', moves: ['meanlook']}]);
+		battle.join('p2', 'Guest 2', 1, [
+			{species: "Trevenant", ability: 'naturalcure', item: 'shedshell', moves: ['ingrain']},
```

Comment:
Thanks for pointing that out. It also allowed me to notice that I straight up forgot to set the test to actually have the Pokemon move once with Mean Look / Ingrain before trying to switch